tutorials/llm/llama-3/nemo2-sft-peft/nemo2-peft.ipynb

```diff
@@ -0,0 +1,540 @@
+    "We use the `llm.import_ckpt` API to download the specified model using the \"hf://<huggingface_model_id>\" URL format. It will then convert the model into NeMo 2.0 format. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import nemo_run as run\n",
+    "from nemo import lightning as nl\n",
+    "from nemo.collections import llm\n",
+    "from megatron.core.optimizer import OptimizerConfig\n",
+    "from nemo.collections.llm.peft.lora import LoRA\n",
+    "import torch\n",
+    "import pytorch_lightning as pl\n",
+    "from pathlib import Path\n",
+    "from nemo.collections.llm.recipes.precision.mixed_precision import bf16_mixed\n",
+    "\n",
+    "\n",
+    "# llm.import_ckpt is the nemo2 API for converting Hugging Face checkpoint to NeMo format\n",
+    "# example usage:\n",
+    "# llm.import_ckpt(model=llm.llama3_8b.model(), source=\"hf://meta-llama/Meta-Llama-3-8B\")\n",
+    "#\n",
+    "# We use run.Partial to configure this function\n",
+    "def configure_checkpoint_conversion():\n",
+    "    return run.Partial(\n",
+    "        llm.import_ckpt,\n",
+    "        model=llm.llama3_8b.model(),\n",
+    "        source=\"hf://meta-llama/Meta-Llama-3-8B\",\n",
+    "        overwrite=False,\n",
+    "    )\n",
+    "\n",
+    "# configure your function\n",
+    "import_ckpt = configure_checkpoint_conversion()\n",
+    "# define your executor\n",
+    "local_executor = run.LocalExecutor()\n",
+    "\n",
+    "# run your experiment\n",
+    "run.run(import_ckpt, executor=local_executor)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 2: Prepare data\n",
+    "\n",
+    "We will be using SQUAD for this notebook. NeMo 2.0 already provides a `SquadDataModule`. Example usage:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def squad() -> run.Config[pl.LightningDataModule]:\n",
+    "    return run.Config(llm.SquadDataModule, seq_length=2048, micro_batch_size=1, global_batch_size=8, num_workers=0)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For how to use your own data to create your custom `DataModule` in order to perform PEFT, refer to [NeMo 2.0 SFT notebook](./nemo2-sft.ipynb)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 3: Run PEFT with NeMo 2.0 API and NeMo-Run\n",
+    "\n",
+    "The following python script utilizes NeMo 2.0 API to perform PEFT. In this script we are configuring the following components for training. These components are similar between SFT and PEFT. SFT and PEFT both uses `llm.finetune` API. To switch from SFT to PEFT you just need to add `peft` with LoRA adapter to the API parameter.\n",
+    "\n",
+    "### Trainer\n",
```

Comment:
suggested heading revision to indicate it is a step in the procedure (that is, add an imperative verb to the heading)

Use Trainer